hab/train.py

```diff
@@ -62,60 +65,37 @@ def train(train_data_dir: str, test_data_dir: str, magnitude_increase: int = 1):
     train_loader = DataLoader(train_dataset)
     test_loader = DataLoader(test_dataset)
 
-    criterion = nn.CrossEntropyLoss()
-    optimizer = optim.Adam(lr=1e-3)
-
     logger.info("Initial Seed: %d" % (torch.initial_seed()))
 
-    # instantiate HABs CNN
-    habs_model = HABsModelCNN()
-
     logger.info("Training model.")
-
-    # train
-    for epoch in range(2):
-        for i, data in enumerate(train_loader, 0):
-            images, targets = data
-
-            optimizer.zero_grad()
-
-            outputs = habs_model(images)  # nn.module __call__()
-            loss = criterion(outputs, targets)
-            loss.backward()
-            optimizer.step()
-
-            running_loss += loss.item()
-            if i % 2000 == 1999:  # print every 2000 mini-batches
-                logger.info("[%d, %5d] loss: %.3f" %
-                            (epoch + 1, i + 1, running_loss / 2000))
-                running_loss = 0.0
+    training_laps(model, train_loader, epochs, optimizer, criterion)
 
     logger.info("Testing model.")
-
-    # test
-    correct = 0
-    total = 0
-    habs_model.eval()
-    with torch.no_grad():
-        for data in test_loader:
-            images, targets = data
-
-            outputs = habs_model(images)  # nn.module __call__()
-            _, predicted = torch.max(outputs.data, 1)
-            total += targets.size(0)
-            correct += (predicted == targets).sum().item()
-
-    logger.info("Accuracy of the network on the 10000 test images: %d %%" % (
-            100 * correct / total))
-
-
-def main(train_dataset: str, test_dataset: str):
+    evaluate(model, test_loader)
```

Comment:
If this is *not* a holdout set, you want to move this thing into the training loop after the batchloader for loops. Think of it sort of like quizzes. After every iteration through the material, you want to quiz your model and see how things are going. Plus this will also let you introduce learn rate optimizers.

Reply:
My original thinking was that the `evaluate` method would consist of a holdout set that would test on the final trained model. In order to test the model as it trains like you mentioned during our call, I created a 3rd dataset, `validation dataset`. This also contains a holdout set, but it is now given after the `training_lap` inside of the training loop. Is adding a `validation lap` what you were talking about here?

Reply:
During our call on 5.5.21 Ty confirmed this was correct.